src/Connection/StreamConnection.php

```diff
@@ -165,6 +169,7 @@ private function read(int $length, string $errorMessage) : string
 
         /** @psalm-suppress PossiblyNullArgument */
         $meta = \stream_get_meta_data($this->stream);
-        throw new CommunicationFailed($meta['timed_out'] ? 'Read timed out' : $errorMessage);
+        throw $meta['timed_out'] ? new CommunicationFailed('Read timed out')
+                                 : CommunicationFailed::withLastPhpError($errorMessage);
```

Comment:
nit: A matter of preference, but I find such formatting with large indents is harder to read. What about
```suggestion
        if ($meta['timed_out']) { 
            throw new CommunicationFailed('Read timed out');
        }

        throw CommunicationFailed::withLastPhpError($errorMessage);
```

Reply:
I don't really have a preference for this case. Changed to your version.